### PR TITLE
fs-extra: Add extra allowed types

### LINKED ISSUE #30398: @types/fs-extra -- Updates needed for Node v10.10+
`@types/fs-extra` needs to be updated to account for changes in Node v10.10+.
Tagging @alan-agius4 @midknight41 @shiftkey @mees- @jrockwood

Relevant bits of the changelog are:

### Notable changes

* **fs**
  * Added a `recursive` option to `fs.mkdir` and `fs.mkdirSync`. If this option
    is set to true, non-existing parent folders will be automatically created. [#21875](https://github.com/nodejs/node/pull/21875)

### Notable Changes

* **fs**:
  * The methods `fs.read`, `fs.readSync`, `fs.write`, `fs.writeSync`,
    `fs.writeFile` and `fs.writeFileSync` now all accept `TypedArray` and
    `DataView` objects. [#22150](https://github.com/nodejs/node/pull/22150)
  * A new boolean option, `withFileTypes`, can be passed to `fs.readdir` and
    `fs.readdirSync`. If set to true, the methods return an array of directory
    entries. These are objects that can be used to determine the type of each
    entry and filter them based on that without calling `fs.stat`. [#22020](https://github.com/nodejs/node/pull/22020)

### PR DESCRIPTION
**Pull request by NotWoods**
Please fill in this template.

- [ ] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

---

Fixes #30398

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## fs-extra ([<kbd>unpkg</kbd>](https://unpkg.com/browse/fs-extra@latest/))
was missing the following properties:
1. emptydirSync
1. emptydir
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e3e0342df680db3f36981b0e392526379d88cece
</p>

File: types/fs-extra/index.d.ts [hunk 1]
```diff
@@ -158,32 +160,32 @@ export function futimes(fd: number, atime: Date, mtime: Date, callback: (err: No
...
-export function mkdir(path: string | Buffer): Promise<void>;
+export function mkdir(path: PathLike, options: Mode | fs.MakeDirectoryOptions | null, callback: (err: NodeJS.ErrnoException) => void): void;
+export function mkdir(path: PathLike): Promise<void>;
 
-export function open(path: string | Buffer, flags: string | number, callback: (err: NodeJS.ErrnoException, fd: number) => void): void;
-export function open(path: string | Buffer, flags: string | number, mode: number, callback: (err: NodeJS.ErrnoException, fd: number) => void): void;
-export function open(path: string | Buffer, flags: string | number, mode?: number): Promise<number>;
+export function open(path: PathLike, flags: string | number, callback: (err: NodeJS.ErrnoException, fd: number) => void): void;
+export function open(path: PathLike, flags: string | number, mode: Mode, callback: (err: NodeJS.ErrnoException, fd: number) => void): void;
+export function open(path: PathLike, flags: string | number, mode?: Mode | null): Promise<number>;
```
Comment (peterblazejewicz) [inline types/fs-extra/index.d.ts:188]:
```suggestion
export function open(path: PathLike, flags: string | number, mode?: Mode | null): Promise<number>;
```
can you point usage/source with this param allowing additional type: null?

Comment (NotWoods) [reply types/fs-extra/index.d.ts:188]: Based on equivalent fs method: https://github.com/NotWoods/DefinitelyTyped/blob/fs-extra-update/types/node/fs.d.ts#L1167

Comment (peterblazejewicz) [reply types/fs-extra/index.d.ts:188]: thank you!

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@NotWoods thx for the contribution!

**Review by rbuckton — APPROVED** [review]

**Comment by NotWoods** [comment]
Ready to merge

**Pull request merged**